### DISCUSSION TITLE
Comparazione sensore a stringa

### DISCUSSION THREAD

**Discussion by evirci** [indomus/forum#2584] [HUB: Home Assistant]
Ciao a tutti, sapreste consigliarmi dove sbaglio?
Ho scritto un codice che dovrebbe comparare lo stato di un sensore con una stringa, ma il risultato booleano risulta errato. Ci sbatto la testa da giorni e non capisco dove sbaglio,

` - condition: template
            value_template: >-
              {{ not is_state("sensor.precipitazioni_notte", "prec.: assenti") }} `

Mi da come risultato vero, ma visto che è negato e la stringa è uguale al sensore dovrebbe darmi falso

![Cattura](https://github.com/indomus/forum/assets/122167724/aa7d0b73-f528-46dd-a129-1c5b67f4834b)

**Comment by snakuzzo** [comment]
Innanzi tutto formatta bene quel codice, che così come hai scritto non si capisce niente.
Assicurati che non ci siano caratteri sporchi.
Io ho appena fatto una prova da dev tools impostando un input_text con quella stringa e torna tutto

```
{{ states('input_text.prova') }}   # Risultato -> prec.: assenti

{{ not is_state('input_text.prova','prec.: assenti') }}   # Risultato -> False

{{ is_state('input_text.prova','prec.: assenti') }}   # Risultato -> True
```

**Reply by evirci** [reply]
ok, credo di aver capito il perchè, molto subdolo...
dallo screenshot che ti allego si evince che il mio sensore.precipitazioni_notte non restituisce solo prec.: assenti, ma in più anche una riga a capo...
ora non potendo modificare cosa mi restituisce quel sensore, è uno scrape, come posso dire che deve essere uguale a prec.: assenti + riga a capo?
![Cattura2](https://github.com/indomus/forum/assets/122167724/ca28c32c-1b21-4e56-a68c-5455017b3477)

**Reply by snakuzzo** [reply]
Nel template che usi per creare il sensore prova ad usare

```
 | replace('\n', '') }}
```

**Reply by evirci** [reply]
l'ho creato da gui, è possibile comunque farlo?

**Reply by snakuzzo** [reply]
https://www.home-assistant.io/integrations/scrape/

**Reply by evirci** [reply]
ok, se con un modello come questo ``` '{{ value | replace('\n', '') | replace(' ', '') }}'``` mi restituisce ```'prec.:assenti'  ``` come tolgo quei apici? ho provato con ``` replace(''', '')``` ma logicamente da errore...inoltre ci sono anche degli spazi dopo, non vorrei fosse un problema

**Reply by snakuzzo** [reply]
Non ho capito cosa hai chiesto.
Se fai come hai scritto rimuovi tutti gli spazi...e non è quello che vuoi.

Comunque usa i doppi apici fuori e i singoli dentro...

```
"{{ value | replace('\n', '') }}"
```

**Reply by evirci** [reply]
con l'esatto modello che hai riportato ottengo ```" prec.: assenti "```
quindi ho elaborato questo ```'{{ value | replace('\n', '') | replace(' ', '') }}'``` per togliere gli spazi prima prec e dopo assenti, ma rimangono gli apici estremi senza contare che ho tolto pure lo spazio fra : assenti

**Reply by evirci** [reply]
Ho risolto, che tonto!

**Reply by snakuzzo** [reply]
Ho letto ora.
Ok, meglio così...mi era sfuggito che avevi anche gli apici doppi all'interno

**Reply by snakuzzo** [reply]
Comunque, per la cronaca si fa così... e se vuoi togliere gli spazi prima e dopo si usa `trim`

```
"{{ value | replace('\n', '') | replace('"','') | trim }}"
```

**Reply by evirci** [reply]
grazie mille per l'aiuto!!!
